pkg/controllers/openshift-crd-manager.go

```diff
@@ -47,5 +47,10 @@ func (s *OpenShiftCRDManager) Run(ctx context.Context, ready chan<- struct{}, st
 	}
 	logrus.Infof("%s applied default CRDs", s.Name())
 
+	logrus.Infof("%s waiting for CRDs acceptance before proceeding", s.Name())
+	if err := assets.WaitForCrdsEstablished(s.cfg); err != nil {
+		logrus.Errorf("%s unable to confirm all CRDs are ready: %v", s.Name(), err)
+	}
+	logrus.Infof("%s all CRDs are ready", s.Name())
```

Comment:
should it exit here on error?

Reply:
I was scratching my head a bit about the intent in the original code, which just logs the error, signals the service is "ready" unconditionally after it returns.  I opted to keep the PR scope focused on the wait logic.

https://github.com/redhat-et/microshift/blob/c006b98aa6003d14f13ba68d5341004b85c85e39/pkg/controllers/openshift-crd-manager.go#L42-L51